File: warmup/7_email_address.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
import re
line=input()
data=''

r=[]
for _ in range(line):
    d=raw_input()
    regex=r'\b[\w\-][\w\-\.]+@[\w\-][\w\-\.]+[a-zA-Z]{1,4}\b'
    all=re.findall(regex,d)
    r=r+all
print ';'.join(sorted(set(r)))

File: warmup/7_email_address.pl
use v5.14;
use warnings;

my $nol = <> + 0;

my $r = '\b[\w\-][\w\-\.]+@[\w\-][\w\-\.]+[a-zA-Z]{1,4}\b';
my $regex = quotemeta($r);
my %mail;
my @final;

foreach (1 .. $nol) {
    my $line = <>;
    chomp($line);

    if($line =~ /($r)/){
        $mail{$1} += 1;
    } 
}

foreach my $key (sort {$a cmp $b} keys %mail) {
    push @final, $key;
}

my $output = join(';', @final);
say $output;
